### ISSUE TITLE
🐛 FromFactory customization on enums produces incorrect values

### ISSUE THREAD

**Issue by Quogu** [AutoFixture/AutoFixture#1372]
### Describe the Bug
Adding a `FromFactory` customization to a fixture to specify the value of an enum does not use the value from the customization, but instead appears to produce random values of the enum's underlying type.

### Scenario
```
using System;
using AutoFixture;
					
public class Program
{
	public static void Main()
	{
		FixtureCreateStringFromFactory();
		FixtureCreateEnumNoCustomization();
		FixtureCreateEnumFromFactory();
	}
	
	public static void FixtureCreateStringFromFactory()
	{
		var fixture = new Fixture();
		fixture.Customize<string>(c => c.FromFactory(() => "hello"));
		var items = fixture.CreateMany<string>();
		foreach (var item in items)
		{
			// Notice how all values are "hello".
			Console.WriteLine(item);
		}
		Console.WriteLine("");
	}
	
	public enum TestEnum
	{
		A = 0,
		B = 1,
		C = 2,
		D = 3,
	}
	
	public static void FixtureCreateEnumNoCustomization()
	{
		var fixture = new Fixture();
		var items = fixture.CreateMany<TestEnum>();
		foreach (var item in items)
		{
			// Notice how all values are valid members of this enum.
			Console.WriteLine(item);
		}
		Console.WriteLine("");
	}
	
	public static void FixtureCreateEnumFromFactory()
	{
		var fixture = new Fixture();
		fixture.Customize<TestEnum>(c => c.FromFactory(() => TestEnum.B));
		var items = fixture.CreateMany<TestEnum>();
		foreach (var item in items)
		{
			// Notice how all values are random byte values.
			Console.WriteLine(item);
		}
		Console.WriteLine("");
	}
}
```

This was ran using https://dotnetfiddle.net/ with the latest AutoFixture package, 4.17.0.

### Expected Behavior
Calls to `fixture.Create<TestEnum>()` would produce the value `TestEnum.B` as specified by the factory.

### Tasks

- [x] Updated all AutoFixture and related libraries
- [ ] Confirmed in support forums if behavior is expected

**Comment by Ergamon** [comment]
I am not so sure, I understand your code and what you want to do.

If you want to pin a value in AutoFixture the correct method is Inject.
```
    var fixture = new Fixture();
    fixture.Inject("Hello");
    fixture.Inject(TestEnum.B);
    var testString = fixture.Create<string>();
    var testEnum = fixture.Create<TestEnum>();
    Console.WriteLine(testString);
    Console.WriteLine(testEnum);
```

There is no difference between an enum and a string. This works for any type.

What you are showing in the code is customization. Basically you can give AutoFixture instead of a fixed value, some generation code. By that you can have a different value every time the type is created.

To make it easier to understand for you, run the following code:
```
    var fixture = new Fixture();

    var testString = fixture.Build<string>().FromFactory(() => "hello").Create();
    Console.WriteLine(testString);

    var testEnum = fixture.Build<TestEnum>().FromFactory(() => TestEnum.B).Create();
    Console.WriteLine(testEnum);
```

You will see a different output for the string and the enum this time. Basically AutoFixture does a "new" on the type via Activator.CreateInstance.

The FromFactory is a helper to pass in parameters. As string has constructor with a string, you will always get the same value. The enum has no such constructor, so your FromFactory does basically nothing.

**Comment by aivascu** [comment]
@Ergamon I think @Quogu is expecting the resulting customization for the enum to behave as if the value was injected/registered. The difference is that when the value is injected there is no more post-processing to set any public members, and when the value is built using a factory (by default) it will later be picked up by the post processor and it's public members will be set unless they are omitted in the customization.

What does this have to do with the `enum`? As it turns out enums in C# are structures that have a public non-readonly field called `value__`, and AutoFixture is picking up that field and sets it to a new random value based on the underlying type of the `enum`.
![image](https://user-images.githubusercontent.com/7030530/208775597-3a83efa9-657e-4ee5-bc6f-400a8d4d6271.png)

So for instance the following test will produce a result similar to the one of `Inject<T>(T)`.
```cs
[Fact]
public void Foo()
{
    var fixture = new Fixture();
    fixture.Customize<TestEnum>(c => c.FromFactory(() => TestEnum.B)
        .OmitAutoProperties());

    var actual = fixture.CreateMany<TestEnum>().ToList();

    Assert.All(actual, x => Assert.Equal(TestEnum.B, x));
}
```
So basically AutoFixture functions as intended, and this  is just a quirk of the language in how it implements a specific structure.

The question is now, should the enums receive special treatment and behave as injected when constructed using `Customize<T>()`? In my opinion they shouldn't since no one should be freezing enums using `FromFactory` in the first place. There are more than enough ways to do that already.

What do you guys think?

**Comment by aivascu** [comment]
Since there was no reply, I'll consider that there are sufficient alternatives to freeze an `enum` to avoid the `FromFactory` approach.
Will consider a fix if this becomes an issue, or if there will be a significant scenario where this behavior is a problem.
Closing the issue.
